js/classes/entities/midground/weapon/sword.js

```diff
@@ -1,224 +1,114 @@
-    /** The size of the Sword after the scale factor is applied. */
-    static get SCALED_SIZE() {
-        return Vector.multiply(Sword.SIZE, Sword.SCALE);
-    };
-
-    /** The maximum distance the Sword can be from Chad while jabbing before it begins retracting. */
-    static get MAX_OFFSET() {
-        return 30;
-    }
-
-    /** The offset from Chad's y-position applied to the Sword's y-position. */
-    static get Y_OFFSET() {
-        return 10;
-    }
-
     /** The spritesheet containing all the Sword types. */
     static get SPRITESHEET() {
+        // TODO: determine if this property still belongs here. The HUD still uses
+        // it, but it is no longer used within this class.
         return "./sprites/swords.png";
     };
 
-    /** How fast the Sword moves while jabbing. */
-    static get JAB_SPEED() {
-        return 300;
-    };
-
-    /** How fast the Sword moves while retracting. */
-    static get RETRACT_SPEED() {
-        return 150;
-    };
-
-    /** Sword type 1 (rename later). */
-    static get TYPE_1() {
-        return 0;
-    }
-
-    /** Sword type 2 (rename later). */
-    static get TYPE_2() {
-        return 1;
-    }
-
-    /** Sword type 3 (rename later). */
-    static get TYPE_3() {
-        return 2;
-    }
-
-    /** Sword type 4 (rename later). */
-    static get TYPE_4() {
-        return 3;
+    /** The amount of damage the Sword deals during an attack. */
+    static get DAMAGE() {
+        return 5;
     }
 
-    /** Sword type 5 (rename later). */
-    static get TYPE_5() {
-        return 4;
+    /** The duration of an attack, in seconds. */
+    static get DURATION() {
+        return 2 / 5;
     }
 
-    /** The property table for Sword types. */
-    static get PROPERTY_TABLE() {
-        return {
-            [Sword.TYPE_1]: {
-                DAMAGE: 5,
-                DELAY: 0.9
-            },
-            [Sword.TYPE_2]: {
-                DAMAGE: 10,
-                DELAY: 0.6
-            },
-            [Sword.TYPE_3]: {
-                DAMAGE: 20,
-                DELAY: 0.6
-            },
-            [Sword.TYPE_4]: {
-                DAMAGE: 40,
-                DELAY: 0.3
-            },
-            [Sword.TYPE_5]: {
-                DAMAGE: 80,
-                DELAY: 0.0
-            }
-        };
-    };
-
-    /**
-     * Helper method that checks the validity of the provided Sword type.
+    /** 
+     * Whether or not an attack is ongoing. 
      * 
-     * @param {number} type the type value to check
-     * @throws {Error} Will throw an error if type is invalid.
+     * @returns {boolean} true if a Sword attack is underway, false otherwise
      */
-    static checkType(type) {
-        if (typeof type !== "number" || type % 1 !== 0 || type < 0 || type > 4) {
-            throw new Error("Invalid Sword type: please use a Sword member type (e.g. Sword.TYPE_1).");
-        }
+    isSlicing() {
+        return (Date.now() - this.lastAttack) / 1000 < Sword.DURATION;
```

Comment:
This feels a little confusing. Can you explain what is happening? It feels a bit arbitrary.

Reply:
Perhaps you could also JSDoc it.

Reply:
It has JSDoc, look above the deleted `checkType` method. It's not really arbitrary, it just returns whether an attack is currently in progress.